code/game/objects/items/devices/radio/detpack.dm

```diff
@@ -0,0 +1,323 @@
+/obj/item/device/radio/detpack
+	name = "detonation pack"
+	desc = "Programmable remotely triggered 'smart' explosive used for demolitions and impromptu booby traps. Can be set to breach or demolition detonation patterns."
+	gender = PLURAL
+	icon = 'icons/obj/det.dmi'
+	icon_state = "detpack_off"
+	item_state = "plasticx"
+	flags_item = NOBLUDGEON
+	frequency = 1457
+	w_class = 2.0
+	origin_tech = "syndicate=2"
+	on = FALSE
+	layer = MOB_LAYER - 0.1
+	var/armed = FALSE
+	var/timer = 10.0
+	var/code = 2
+	var/det_mode = FALSE //FALSE for breach, TRUE for demolition.
+	var/atom/plant_target = null //which atom the detpack is planted on
+	var/target_drag_delay = null //store this for restoration later
+
+/obj/item/device/radio/detpack/examine(mob/user)
+	. = ..()
+	var/details
+	if(timer)
+		details += "Its timer has [timer] seconds left."
+	if(det_mode)
+		details += " It appears set to demolition mode."
+	else
+		details += " It appears set to breaching mode."
+	if(armed)
+		details += " <b>It is armed!</b>"
+	to_chat(user, "<span class='warning'>[details]</span>")
+
+
+/obj/item/device/radio/detpack/Dispose()
+	if(plant_target && !plant_target.loc)
+		new /obj/item/device/radio/detpack(get_turf(src)) //in the event we have a target that can prematurely get deleted, like a roller bed that's folded up.
+	processing_timers.Remove(src)
+	nullvars()
+	. = ..()
+
+/obj/item/device/radio/detpack/update_icon()
+	if(!plant_target)

```

Comment:
Simplify these sorts of logic by making it so you only need to check each var once. It makes it a lost easier to read and modify later.
```
icon_state = "detpack_[plant_target ? "set_" : ""]"
if(on)
  icon_state = "[icon_state][armed ? "armed" : "on"]"
else
  icon_state = "[icon_state]off"
```
If you don't like ternary that's more of a stylistic choice. Adding on to the string by referencing it with `[]` instead of using `+` is faster though.